src/Smearing.jl

```diff
@@ -43,10 +43,10 @@ function occupation_divided_difference(S::SmearingFunction, x, y, εF, temp)
     else
         f(z) = occupation(S, (z-εF) / temp)
         fder(z) = occupation_derivative(S, (z-εF)/temp) / temp
-        divided_difference_(f, fder, x, y)
+        _divided_difference(f, fder, x, y)
```

Comment:
Here and at other places: why? _ are for very internal "do not touch outside of immediate context" functions, these look generally useful

Reply:
Don't care too much in this case. I think it was even @antoine-levitt who chose the name :smile:.

Reply:
Also we should really start a document where we list these conventions (i.e. in this case: "Prefix function name by `_` if it is an internal helper function, which is not of general use and should thus be kept close to the vicinity of the calling function")